DB/CreatingBTdb.sql

```diff
@@ -0,0 +1,171 @@
+)
+
+
+CREATE TABLE TransitRates
+(
+	ID tinyint IDENTITY,
+	BeginDate date NOT NULL,
+	EndDate date NUll,
+	Rate smallMoney NOT NULL
+)
+
+
+CREATE TABLE DailyRates
+(
+	ID tinyint IDENTITY,
+	BeginDate date NOT NULL,
+	EndDate date NUll,
+	Rate smallMoney NOT NULL
+)
+
+
+CREATE TABLE DwellingRates
+(
+	ID tinyint IDENTITY,
+	LocalityTypeId tinyint FOREIGN KEY REFERENCES LocalityTypes (ID),	
+	BeginDate date NOT NULL,
+	EndDate date NUll,
+	Rate smallMoney NOT NULL
+)
+
+
+CREATE TABLE Staff
+(
+	ID smallint CONSTRAINT PK_EmployeeID PRIMARY KEY IDENTITY,
+	RankID tinyint NULL,
+	FirstName nvarchar(30) NOT NULL,
+	LastName nvarchar(30)NOT NULL,
+	Patronymic nvarchar(30) NULL,
+	UserID tinyint NOT NULL,
+	CONSTRAINT FK_RankID FOREIGN KEY (RankID) REFERENCES Ranks (ID),
+	CONSTRAINT FK_UserID FOREIGN KEY (UserID) REFERENCES Users (ID)
+)
+
+CREATE TABLE BusinessTrips
+(
+	ID int PRIMARY KEY IDENTITY,
+	EmployeeID smallint FOREIGN KEY REFERENCES Staff (ID) NOT NULL,
+	Info nchar(300) NULL,
+	BusinessTripState tinyint NOT NULL
+)
+
+
+CREATE TABLE OrderInfo
+(
+	ID smallint IDENTITY PRIMARY KEY,
+	BusinessTripId int FOREIGN KEY REFERENCES BusinessTrips (ID) UNIQUE,
+	OrderNumber smallint NOT NULL,
+	OrderDate date NOT NULL,
+	StartDate date NOT NULL,
+	EndDate date NOT NULL,
+	LocalityId tinyint FOREIGN KEY REFERENCES Localities (ID) NOT NULL,
+	Permanent bit NOT NULL,
+	TrafficForward bit NOT NULL,
+	TrafficBack bit NOT NULL,
+	Feeding bit NOT NULL,
+	Transport bit NOT NULL,
+	Lodging bit NOT NULL,
+	RationPack tinyint NOT NULL,
+	TeamId smallint NOT NULL,
+	PurposeId tinyint NOT NULL
+)
+
+
+CREATE TABLE ChangedOrderInfo
+(	
+	OrderInfoId smallint FOREIGN KEY REFERENCES OrderInfo (ID) UNIQUE,
+	OrderNumber smallint NOT NULL,
+	OrderDate date NOT NULL,
+	StartDate date NULL,
+	EndDate date NULL,
+	LocalityId tinyint FOREIGN KEY REFERENCES Localities (ID) NULL,
+	Permanent bit NULL,
+	TrafficForward bit NULL,
+	TrafficBack bit NULL,
+	Feeding bit NULL,
+	Transport bit NULL,
+	Lodging bit NULL,
+	RationPack tinyint NULL,
+	PurposeId tinyint NULL
+)
+
+
+CREATE TABLE StatementInfo
+(	
+	BusinessTripId int FOREIGN KEY REFERENCES BusinessTrips (ID) UNIQUE,
+	ChangeOrderNumber smallint NOT NULL,
+	StartDate date NOT NULL,
+	EndDate date NOT NULL,
+	LocalityId tinyint FOREIGN KEY REFERENCES Localities (ID) NOT NULL,
+	Permanent bit NOT NULL,
+	TrafficForward bit NOT NULL,
+	TrafficBack bit NOT NULL,
+	Feeding bit NOT NULL,
+	Transport bit NOT NULL,
+	Lodging bit NOT NULL,
+	RationPack tinyint NOT NULL,
+	TeamId smallint NOT NULL
+)
+
+
+CREATE TABLE Calculations
+(
+	ID smallint PRIMARY KEY IDENTITY,
+	BusinessTripID int FOREIGN KEY REFERENCES BusinessTrips (ID)NOT NULL,
+	TransitCosts smallmoney NULL,
+	DailyCosts smallmoney NULL,
+	DwellingCosts smallmoney NULL,
+	PaySheetNumber tinyint NOT NULL,
+	PaySheetDate date NOT NULL
+)
+
+ALTER TABLE Calculations
```

Comment:
Why can't we make these alterations during CREATE TABLE?

Reply:
Roger